File: frontend/src/types/candidate.ts
export interface Candidate {
  id: number;
  full_name: string;
  email: string;
  phone?: string;
  skills: string;
  experience_years: number;
  status: string;
  score: number;
  score_breakdown: {
    strengths: string[];
    weaknesses: string[];
    details: Record<string, string>;
  };
  semantic_score?: number;
  match_signals?: {
    matched_skills?: string[];
    missing_skills?: string[];
    skill_gap_analysis?: Array<{
      skill: string;
      severity: "Critical" | "Preferred";
      reason: string;
    }>;
    strengths?: string[];
    weaknesses?: string[];
    interview_focus_areas?: string[];
  };
  job_id: number;
  job_title?: string;
  file_path: string;
  raw_text?: string;
  current_org?: string;
  notice_period?: string;
  moodle_user_id?: number | null;
  assessment_status?: string;
  assessment_score?: number | null;
  assessment_percentage?: number | null;
  assessment_assigned_at?: string | null;
  assessment_completed_at?: string | null;
}

export interface CandidateNote {
  id: number;
  candidate_id: number;
  recruiter_id: number;
  comment: string;
  created_at: string;
}

File: backend/app/schemas/candidate.py
from datetime import datetime
from typing import Optional, Dict, Any, Literal

from pydantic import BaseModel, Field

AssessmentStatus = Literal[
    "NOT_ASSIGNED",
    "ASSIGNED",
    "IN_PROGRESS",
    "COMPLETED",
    "PASSED",
    "FAILED",
]

class CandidateBase(BaseModel):
    full_name: Optional[str] = None
    email: Optional[str] = None
    phone: Optional[str] = None
    skills: Optional[str] = None
    education: Optional[str] = None
    experience_years: int = 0
    current_org: Optional[str] = None
    notice_period: Optional[str] = None
    status: str = "Applied"

class CandidateCreate(CandidateBase):
    job_id: int
    file_path: str
    raw_text: Optional[str] = None

class Candidate(CandidateBase):
    id: int
    job_id: int
    job_title: Optional[str] = None
    score: float
    score_breakdown: Optional[Dict[str, Any]] = None
    semantic_score: Optional[float] = None
    match_signals: Optional[Dict[str, Any]] = None
    reapplication_details: Optional[Dict[str, Any]] = None
    file_path: Optional[str] = None
    raw_text: Optional[str] = None
    moodle_user_id: Optional[int] = None
    moodle_username: Optional[str] = None
    moodle_temp_password: Optional[str] = None
    assessment_status: AssessmentStatus = Field(default="NOT_ASSIGNED")
    assessment_score: Optional[float] = None
    assessment_percentage: Optional[float] = None
    assessment_assigned_at: Optional[datetime] = None
    assessment_completed_at: Optional[datetime] = None

    class Config:
        from_attributes = True
